modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/RESTBuilder.java

```diff
@@ -2087,53 +1985,7 @@ private String _getSchemaVarName(
 		return freeMarkerTool.getSchemaVarName(reference.substring(index + 1));
 	}
 
-	private void _invokeClientJSGenerator(
-			File baseClientJSDir, File openAPIYAMLFile, String targetClientType)
-		throws Exception {
-
-		String outputPathString = StringBundler.concat(
-			baseClientJSDir.getPath(), "/src/", targetClientType);
-
-		ProcessBuilder processBuilder = new ProcessBuilder(
-			Arrays.asList(
-				_getNPMPathString(), "exec", "--prefix", _getNodePrefix(),
-				"--yes", "@openapitools/openapi-generator-cli@2.15.3", "--",
-				"generate",
-				"--additional-properties=modelPropertyNaming=original," +
-					"paramNaming=original",
-				"--generator-name", "typescript-" + targetClientType,
-				"--global-property",
-				"apis,models,supportingFiles=api.ts:apis.ts:models.ts",
-				"--input-spec", openAPIYAMLFile.getPath(), "--output",
-				outputPathString, "--skip-validate-spec"));
-
-		String nodeBinPathString = _getNodeBinPathString();
-
-		if (nodeBinPathString != null) {
-			_prependPath(processBuilder, nodeBinPathString);
-		}
-
-		Process process = processBuilder.start();
-
-		process.waitFor();
-
-		if (process.exitValue() > 0) {
-			Scanner scanner = new Scanner(process.getErrorStream());
-
-			scanner.useDelimiter("\n");
-
-			while (scanner.hasNext()) {
-				System.out.println(
-					"Unable to generate client JS: " + scanner.next());
-			}
-		}
-
-		Files.deleteIfExists(Paths.get("./openapitools.json"));
-
-		_deleteDir(Paths.get(outputPathString, ".openapi-generator"));
-	}
-
-	private void _invokeClientJSGenerator(String openAPIYAMLString)
+	private void _invokeClientTSGenerator(String openAPIYAMLString)
 		throws Exception {
 
 		File baseClientJSDir = new File(
```

Comment:
Move all these inside the typescript generator and remove this method, just call the generator in the calling method.